File: remove_duplicate_words/solution_2.js
function removeDuplicateWords (s) {
  let words = s.split(" ");
  let str = "";
  
  for (let word of words) {
    if (!words[word]) {
      words[word] = word;
      str += word + " ";
    }
  }
  
  return str.slice(0, str.length - 1);
}

File: solutions/shortest_word/solution_2.go
package kata

import "strings"

func FindShort(s string) int {
  words := strings.Split(s, " ")
  shortest := len(s)
  
  for _, word := range words {
    if (len(word) < shortest) {
      shortest = len(word)
    }
  }
  
  return shortest
}